src/Workspaces/VisualBasic/Portable/Rename/VisualBasicRenameRewriterLanguageService.vb

```diff
@@ -275,6 +277,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Rename
                             suffix = "Event"
                         End If
 
+                        If fieldSymbol.AssociatedSymbol IsNot Nothing AndAlso
+                           fieldSymbol.AssociatedSymbol.IsKind(SymbolKind.Property) AndAlso
+                           fieldSymbol.Name = "_" + fieldSymbol.AssociatedSymbol.Name Then
```

Comment:
If there was a future meaning of AssociatedSymbol that was also a property, we'd probably have to add a better distinguisher (AssociatedSymbolReason) or something versus expect consumers of the API to do checks like this.